src/SuperSocket.WebSocket/WebSocketPipelineFilter.cs

```diff
@@ -1,146 +1,89 @@
-            {
-                items = null;
-                return false;
-            }
-
-            items = new NameValueCollection();
-
-            var prevKey = string.Empty;
-            var line = string.Empty;
-            
-            while (!string.IsNullOrEmpty(line = reader.ReadLine()))
+            if (headerLine.IsSingleSegment)
             {
-                if (line.StartsWith(_TAB) && !string.IsNullOrEmpty(prevKey))
-                {
-                    var currentValue = items.Get(prevKey);
-                    items[prevKey] = currentValue + line.Trim();
-                    continue;
-                }
-
-                int pos = line.IndexOf(_COLON);
-
-                if (pos <= 0)
-                    continue;
-
-                string key = line.Substring(0, pos);
-
-                if (!string.IsNullOrEmpty(key))
-                    key = key.Trim();
-
-                if (string.IsNullOrEmpty(key))
-                    continue;
-
-                var valueOffset = pos + 1;
-
-                if (line.Length <= valueOffset) //No value in this line
-                    continue;
-
-                var value = line.Substring(valueOffset);
-
-                if (!string.IsNullOrEmpty(value) && value.StartsWith(' ') && value.Length > 1)
-                    value = value.Substring(1);
-
-                var existingValue = items.Get(key);
-
-                if (string.IsNullOrEmpty(existingValue))
-                {
-                    items.Add(key, value);
-                }
-                else
-                {
-                    items[key] = existingValue + ", " + value;
-                }
-
-                prevKey = key;
+                var span = headerLine.FirstSpan;
+                var colon = span.IndexOf((byte)':');
+                headerName = span.Slice(0, colon);
+                headerValue = span.Slice(colon + 1);
             }
-
-            return true;
-        }
-
-        protected virtual HttpHeader CreateHttpHeader(string verbItem1, string verbItem2, string verbItem3, NameValueCollection items)
-        {
-            return HttpHeader.CreateForRequest(verbItem1, verbItem2, verbItem3, items);
-        }
-
-        private HttpHeader ParseHttpHeaderItems(ref ReadOnlySequence<byte> header)
-        {
-            if (!TryParseHttpHeaderItems(ref header, out var firstLine, out var items))
-                return null;
-
-            var verbItems = firstLine.Split(' ', 3);
-
-            if (verbItems.Length < 3)
+            else
             {
-                // invalid first line
-                return null;
+                var headerReader = new SequenceReader<byte>(headerLine);
+                headerReader.TryReadTo(out headerName, (byte)':');
+                var remaining = headerReader.Sequence.Slice(headerReader.Position);
+                headerValue = remaining.IsSingleSegment ? remaining.FirstSpan : remaining.ToArray();
```

Comment:
There should be a way to read it as string directly without ToArray()

Reply:
<img width="838" alt="image" src="https://github.com/kerryjiang/SuperSocket/assets/39239954/6e325bd6-f0e3-458c-9643-b48cad7893cc">